asapdiscovery-docking/asapdiscovery/docking/workflows/large_scale_docking.py

```diff
@@ -0,0 +1,359 @@
+    data_intermediates = Path(output_dir / "data_intermediates")
+    data_intermediates.mkdir(exist_ok=True)
+
+    if inputs.postera:
+        # load postera
+        logger.info(
+            f"Loading Postera database molecule set {inputs.postera_molset_name}"
+        )
+        postera_settings = PosteraSettings()
+        postera = PosteraFactory(
+            settings=postera_settings, molecule_set_name=inputs.postera_molset_name
+        )
+        query_ligands = postera.pull()
+    else:
+        # load from file
+        logger.info(f"Loading ligands from file: {inputs.filename}")
+        molfile = MolFileFactory.from_file(inputs.filename)
+        query_ligands = molfile.ligands
+
+    # load complexes from a directory or from fragalysis
+    if inputs.structure_dir:
+        logger.info(f"Loading structures from directory: {inputs.structure_dir}")
+        structure_factory = StructureDirFactory.from_dir(inputs.structure_dir)
+        complexes = structure_factory.load(
+            use_dask=inputs.use_dask, dask_client=dask_client
+        )
+    else:
+        logger.info(f"Loading structures from fragalysis: {inputs.fragalysis_dir}")
+        fragalysis = FragalysisFactory.from_dir(inputs.fragalysis_dir)
+        complexes = fragalysis.load(use_dask=inputs.use_dask, dask_client=dask_client)
+
+    # prep complexes
+    logger.info("Prepping complexes")
+    prepper = ProteinPrepper(du_cache=inputs.du_cache)
+    prepped_complexes = prepper.prep(
+        complexes, use_dask=inputs.use_dask, dask_client=dask_client
+    )
+    del complexes
+
+    if inputs.gen_du_cache:
+        logger.info(f"Generating DU cache at {inputs.gen_du_cache}")
+        prepper.cache(prepped_complexes, inputs.gen_du_cache)
+
+    # define selector and select pairs
+    # using dask here is too memory intensive as each worker needs a copy of all the complexes in memory
+    # which are quite large themselves, is only effective for large numbers of ligands and small numbers of complexes
+    logger.info("Selecting pairs for docking based on MCS")
+    selector = MCSSelector()
+    pairs = selector.select(
+        query_ligands,
+        prepped_complexes,
+        n_select=inputs.n_select,
+        use_dask=False,
+        dask_client=None,
+    )
+
+    del prepped_complexes
+
+    # dock pairs
+    logger.info("Running docking on selected pairs")
+    docker = POSITDocker()
+    results = docker.dock(
+        pairs,
+        use_dask=inputs.use_dask,
+        dask_client=dask_client,
+    )
+
+    del pairs
+
+    # write docking results
+    logger.info("Writing docking results")
+    POSITDocker.write_docking_files(results, output_dir / "docking_results")
+
+    # score results
+    logger.info("Scoring docking results")
+    scorer = MetaScorer(
+        scorers=[
+            ChemGauss4Scorer(),
+            GATScorer.from_latest_by_target(inputs.target),
+            SchnetScorer.from_latest_by_target(inputs.target),
+        ]
+    )
```

Comment:
Do we want to give the user control over the scorers used and how the poses are then ranked?

Reply:
Yes to which scorers, for the meantime lets leave the pose ranking as we haven't established a robust protocol for ML ranking.